plugins/modules/idrac_reset.py

```diff
@@ -3,64 +3,147 @@
 
 #
 # Dell OpenManage Ansible Modules
-# Version 7.1.0
-# Copyright (C) 2018-2022 Dell Inc. or its subsidiaries. All Rights Reserved.
+# Version 9.2.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
 
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 #
 
-
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
-DOCUMENTATION = """
+DOCUMENTATION = r'''
 ---
 module: idrac_reset
-short_description: Reset iDRAC
-version_added: "2.1.0"
+short_description: Factory reset the iDRACs
+version_added: "9.2.0"
 description:
-    - This module resets iDRAC.
-    - "iDRAC is not accessible for some time after running this module. It is recommended to wait for some time,
-    before trying to connect to iDRAC."
+  - This module resets the iDRAC to factory default settings.
 extends_documentation_fragment:
   - dellemc.openmanage.idrac_auth_options
+options:
+  reset_to_default:
+    type: str
+    description:
+        - If this value is not set the default behaviour is to restart the iDRAC.
+        - C(All) Discards all settings and reset to default credentials.
+        - C(ResetAllWithRootDefaults) Discards all settings and reset the default username to root and password to the shipping value.
+        - C(Default) Discards all settings, but preserves user and network settings.
+        - C(CustomDefaults) All configuration is set to custom defaults.This option is supported on firmware version 7.00.30.00 and newer versions.
+    choices: ['Default', 'All', 'ResetAllWithRootDefaults', 'CustomDefaults']
+    version_added: 9.2.0
+  custom_defaults_file:
+    description:
+      - Name of the custom default configuration file in the XML format.
+      - This option is applicable when I(reset_to_default) is C(CustomDefaults).
+      - I(custom_defaults_file) is mutually exclusive with I(custom_defaults_buffer).
+    type: str
+  custom_defaults_buffer:
+    description:
+      - This parameter provides the option to import the buffer input in XML format as a custom default configuration.
+      - This option is applicable when I(reset_to_default) is C(CustomDefaults).
+      - I(custom_defaults_buffer) is mutually exclusive with I(custom_defaults_file).
+    type: str
+  wait_for_idrac:
+    description:
+      - This parameter provides the option to wait for the iDRAC to reset and lifecycle controller status to be ready.
+    type: bool
+    default: true
+  job_wait_timeout:
```

Comment:
Job_wait_timeout is not in BSpec, Please check once and update.

Reply:
Added a comment for updating Bspec